packages/core/src/builtin-tasks/contract.ts

```diff
@@ -0,0 +1,209 @@
+  .addParam('privateKey', '', '', paramTypes.string, false)
+  .setAction(async ({ privateKey }: ImportArgs) => {
+    const lockArg = key.privateKeyToBlake160(privateKey)
+
+    const res = execSync('ckb-cli account list --output-format json').toString()
+    const accountList: Array<{ lock_arg: string }> = JSON.parse(res)
+
+    if (accountList.some((account) => account.lock_arg === lockArg)) {
+      return
+    }
+
+    fs.writeFileSync('./.tmp_pk', privateKey)
+
+    console.log('add the user to ckb-cli, enter a password for the signature please')
+    execSync('ckb-cli account import --privkey-path ./.tmp_pk', { stdio: 'inherit' })
+
+    fs.rmSync('./.tmp_pk')
+  })
+
+interface BuildArgs {
+  name?: string
+  release?: boolean
+}
+
+subtask('contract:build')
+  .addParam('name', 'contract name', '', paramTypes.string, true)
+  .addParam('release', 'Build contracts in release mode', false, paramTypes.boolean, true)
+  .setAction(async ({ name, release }: BuildArgs, { run }) => {
+    const workspace = await run('contract:get-workspace')
+    execSync(`cd ${workspace} && capsule build${name ? ` --name ${name}` : ''}${release ? ' --release' : ''}`, {
+      stdio: 'inherit',
+    })
+  })
+
+interface NewArgs {
+  name: string
+  template: string
+}
+
+subtask('contract:new')
+  .addParam('name', 'The name of new contract')
+  .addParam(
+    'template',
+    'language template  [default: rust]  [possible values: rust, c, c-sharedlib]',
+    'rust',
+    paramTypes.string,
+    true,
+  )
+  .setAction(async ({ name, template }: NewArgs, { run }) => {
+    const workspace = await run('contract:get-workspace')
+    console.log('workspace', workspace)
+    execSync(`cd ${workspace} && capsule new-contract ${name} --template ${template}`, { stdio: 'inherit' })
+  })
+
+interface DeployArgs {
+  name?: string
+  chain?: string
+  isMainnet?: boolean
+  fee: number
+  env: string
+  migrate: string
+}
+
+subtask('contract:deploy')
+  .addParam('name', 'name of the contract to be deployed', '', paramTypes.string, true)
+  .addParam(
+    'chain',
+    'ckb rpc url or name [default: dev]  [possible values: dev, test, mainnet, http://localhost:8114 etc...]',
+    'dev',
+    paramTypes.string,
+    true,
+  )
+  .addParam('isMainnet', '', false, paramTypes.boolean, true)
+  .addParam(
+    'fee',
+    "Per transaction's fee, deployment may involve more than one transaction. default: [1]",
+    1,
+    paramTypes.number,
+    true,
+  )
+  .addParam(
+    'env',
+    'Deployment environment. [default: dev]  [possible values: dev, production]',
+    'dev',
+    paramTypes.string,
+    true,
+  )
+  .addParam(
+    'migrate',
+    'Use previously deployed cells as inputs. [default: on]  [possible values: on, off]',
+    'on',
+    paramTypes.string,
+    true,
+  )
+  .setAction(async ({ name, chain, fee, env, migrate, ...args }: DeployArgs, { config, run }) => {
+    const isMainnet = chain === 'mainnet' ? true : args.isMainnet
```

Comment:
How about only using one field to determine `isMainnet`, otherwise the priority should be clear